web/source/kmwuitoggle.ts

```diff
@@ -113,65 +113,71 @@ if(!window['keyman']['ui']['name']) {
     keymanweb['addEventListener']('controlfocused',function(params){ui.doFocus(params.target,true, params['activeControl']);});
     keymanweb['addEventListener']('controlblurred',function(params){ui.doFocus(params.target,false, null);});
 
-    osk['addEventListener']('show',
-      function(oskPosition)
-      {
-        // Ensure that the ui.controller is visible if help is displayed
-        ui.controller.style.display = 'block';
-        ui.oskButton._setSelected(true);
-      
-  /* The following is probably not needed for KMW2, since OSK position is set by KMW, not by the UI        
-        //TODO:  This may not be correct - may need to refer to userLocated argument, not function???
-        // Check if OSK position is user-defined, return if so
-        if (osk['userLocated']()) return oskPosition;
-      
-        // Otherwise, return the position with respect to the focussed element
-        var someElement = keymanweb['getLastActiveElement'](), x, y, w, h, p;
-        if(someElement != null)
+    ui.registerEvents = function() {
+      let osk = keymanweb.osk;
+
+      if(!osk) {
+        return;
+      }
+
+      osk['addEventListener']('show',
+        function(oskPosition)
         {
-          p = util['getAbsolute'](someElement); x = p['x']; y = p['y'];
-          if(someElement.parentWindow && someElement.parentWindow.frameElement)
+          // Ensure that the ui.controller is visible if help is displayed
+          ui.controller.style.display = 'block';
+          ui.oskButton._setSelected(true);
+        
+    /* The following is probably not needed for KMW2, since OSK position is set by KMW, not by the UI        
+          //TODO:  This may not be correct - may need to refer to userLocated argument, not function???
+          // Check if OSK position is user-defined, return if so
```

Comment:
You know... that's a lot of dead code left in the source, there.  Could probably nuke it.